llama.h

```diff
@@ -641,6 +641,10 @@ extern "C" {
     // n_threads_batch is the number of threads used for prompt and batch processing (multiple tokens)
     LLAMA_API void llama_set_n_threads(struct llama_context * ctx, uint32_t n_threads, uint32_t n_threads_batch);
 
+    // Set whether to use causal attention or not
+    // If set to true, the model will only attend to the past tokens
+    LLAMA_API void llama_set_embeddings(struct llama_context * ctx, bool embeddings);
```

Comment:
I'm wondering if it would be more elegant instead of introducing this call, to designate a special token position (for example `llama_batch.pos[i] == INT_MAX`) which indicates that the token attends to and is attended by all other tokens regardless of their positions

This way, no other changes would be necessary and GritLM would simply populate all tokens in the batch with `INT_MAX` and even though it uses causal attention, a non-causal mask could be constructded when necessary

Reply:
That is elegant, but I think we still need the exact position information for positional embeddings, right?

Reply:
Ah yes. So this won't work